sdk/communication/AzureCommunicationCommon/Source/CommunicationIdentifierHelper.swift

```diff
@@ -0,0 +1,89 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import Foundation
+/**
+ Helper class to easily create CommunicationIdentifiers
+ */
+@objcMembers
+public class CommunicationIdentifierHelper: NSObject {
```

Comment:
I'm never a fan of anything that ends in `Helper`, `Util` or `Manager` :-).

@tjprescott Any suggestions for good Swift-idiomatic naming?

Ideally we would have liked a static factory method on `CommunicationIdentifier` itself but can't do it because it's a protocol. Another alternative is a free function.

Reply:
Why not add a required init to the protocol that takes a rawId and then provide the contents of `createCommunicationIdentifier(from:)` as a default implementation via extension?

Reply:
But I agree you don't really want this helper thingy as part of the public API.

Reply:
That was something I was trying. I initially was thinking something like this 
```
@objc public protocol CommunicationIdentifier: NSObjectProtocol {
    @objc var rawId: String { get }
    static func createCommunicationIdentifier(from rawId: String) -> CommunicationIdentifier
}
extension CommunicationIdentifier {
    public static func createCommunicationIdentifier(from rawId: String) -> CommunicationIdentifier {
        
    }
}
```
But because the `CommunicationIdentifier` needed to be exposed to Obj-C I can't create a function inside my extension nicely. 
Otherwise I would be forcing the identifiers to all require this method and every method would do the exact same thing.

Reply:
@tjprescott @DominikMe Had a discussion about this implementation. I think in terms of functionality this approah that we have with the helper most closely aligns with the other platforms. 
That being said, the only reason we are doing this is because of obj-c support. I do think we should slowly start exploring having 2 implementations. One design to favour swift, and this implementation to favour obj-c. We can have in the documentation to guide the developers which approach they should use.